app/views/hyrax/admin/collection_types/_form_settings.html.erb

```diff
@@ -3,11 +3,15 @@
 <p><strong><%= t('.warning') %></strong></p>
 
 <% disabled = (f.object.collections? || f.object.admin_set? || f.object.user_collection?) %>
+<% share_option_disabled = (disabled || !f.object.sharable) %>
```

Comment:
This would be a good thing to put in a form object. Then you could do `f.object.share_option_disabled?`